pandas/io/pytables.py

```diff
@@ -1644,14 +1643,12 @@ def __init__(
 
         self.auto_close = auto_close
 
-    def __iter__(self):
```

Comment:
so there is an open issue to have this just subclass BaseIterator which should preserve the correct iterator behavior. I would rather fix it that way.

Reply:
I don't see the issue you're referring to, but before long I'll do a dedicated look through the HDF5 issues to see what can be closed.

It isn't clear that subclassing BaseIterator would affect the statefulness that this PR is addressing

Reply:
maybe #12953 or #10310

Reply:
umm, not those. but in any event i think if you subclass BaseIterator this will just work

Reply:
do you mean collections.abc.Iterator?  that would mean we'd have to define `__next__` and `__iter__` would return `self`.  But for `__next__` to work, it the relevant `coordinates` variable would have to be stateful, which is exactly what this PR is try to avoid

Reply:
this is a non-standard way of doing things, it needs to inheirt from https://github.com/pandas-dev/pandas/blob/master/pandas/io/common.py#L81

Reply:
Thanks for the link.

If I'm reading you right, you are adamant that `TableIterator` itself should be an iterator.  Is that accurate?